src/System.Drawing.Common/src/System/Drawing/Font.cs

```diff
@@ -275,8 +275,55 @@ public override string ToString()
                                     _gdiCharSet,
                                     _gdiVerticalFont);
         }
-        
+
         // This is used by SystemFonts when constructing a system Font objects.
         internal void SetSystemFontName(string systemFontName) => _systemFontName = systemFontName;
+
+        public unsafe void ToLogFont(object logFont, Graphics graphics)
+        {
+            if (logFont == null)
+            {
+                throw new ArgumentNullException(nameof(logFont));
+            }
+
+            if (Marshal.SizeOf(logFont.GetType()) != sizeof(SafeNativeMethods.LOGFONT))
+            {
+                // If we don't actually have an object that is LOGFONT in size, trying to pass
+                // it to GDI+ is likely to cause an AV.
+                throw new ArgumentException();
+            }
+
+            SafeNativeMethods.LOGFONT nativeLogFont = ToLogFontInternal(graphics);
+            Marshal.PtrToStructure(new IntPtr(&nativeLogFont), logFont);
+        }
+
+        private unsafe SafeNativeMethods.LOGFONT ToLogFontInternal(Graphics graphics)
+        {
+            if (graphics == null)
+            {
+                throw new ArgumentNullException(nameof(graphics));
+            }
+
+            SafeNativeMethods.LOGFONT logFont = new SafeNativeMethods.LOGFONT();
+            Gdip.CheckStatus(Gdip.GdipGetLogFontW(
+                new HandleRef(this, NativeFont), new HandleRef(graphics, graphics.NativeGraphics), ref logFont));
+
+            // Prefix the string with '@' if this is a gdiVerticalFont.
+            if (_gdiVerticalFont)
+            {
+                int fullLength = logFont.lfFaceName.Length;
+                Span<char> temp = stackalloc char[fullLength];
+                temp[0] = '@';
+                logFont.lfFaceName.Slice(0, fullLength - 1).CopyTo(temp.Slice(1));
+                temp.CopyTo(logFont.lfFaceName);
```

Comment:
Span copy operations do handle overlapping buffers. You should not need the extra buffer on stack to do this.

Reply:
Cool, did not know that. I'll update the code.